ffi/src/primitives.rs

```diff
@@ -221,8 +250,29 @@ macro_rules! primitive_impls {
             fn into_ffi(source: &mut [Self]) -> Self::Target {
                 SliceMut::from_slice(source)
             }
-        } )+
-    };
+        }
+
+        impl IntoFfiVec for $ty
+        {
+            type Target = LocalSlice<$ty>;
+
+            fn into_ffi(source: Vec<$ty>) -> Self::Target {
+                source.into_iter().collect()
+            }
+        }
+
+        impl<'itm> TryFromReprCVec<'itm> for $ty {
+            type Source = SliceRef<'itm, $ty>;
+            type Store = ();
+
+            unsafe fn try_from_repr_c(
+                source: Self::Source,
+                _: &'itm mut Self::Store,
+            ) -> Result<Vec<Self>, FfiResult> {
+                source.into_rust().ok_or(FfiResult::ArgIsNull).map(std::borrow::ToOwned::to_owned)
```

Comment:
```suggestion
                source.into_rust().ok_or(FfiResult::ArgIsNull).map(core::borrow::ToOwned::to_owned)
```

Reply:
Oh, i see, you already introduced it in your PR.

Reply:
FYI crate needs to be no_std for the time being